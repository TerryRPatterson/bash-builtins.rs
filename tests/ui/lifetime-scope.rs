use bash_builtins::{Args, BuiltinOptions, Builtin, Result};

#[derive(BuiltinOptions)]
enum Opt<'a> {
    #[opt = 's']
    S(&'a str)
}

struct Foo(&'static str);

impl Builtin for Foo {
    fn call(&mut self, args: &mut Args) -> Result<()> {
        for opt in args.options() {
            match opt? {
                Opt::S(s) => self.0 = s,
            }
        }

        Ok(())
    }
}

fn main() {}
