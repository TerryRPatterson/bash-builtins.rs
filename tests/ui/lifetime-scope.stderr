error[E0759]: `args` has an anonymous lifetime `'_` but it needs to satisfy a `'static` lifetime requirement
  --> $DIR/lifetime-scope.rs:13:25
   |
12 |     fn call(&mut self, args: &mut Args) -> Result<()> {
   |                              --------- this data with an anonymous lifetime `'_`...
13 |         for opt in args.options() {
   |                    ---- ^^^^^^^
   |                    |
   |                    ...is used here...
14 |             match opt? {
15 |                 Opt::S(s) => self.0 = s,
   |                                       - ...and is required to live as long as `'static` here
